cli/module_generate/common/inputs.go

```diff
@@ -42,3 +45,50 @@ type GoModuleTmpl struct {
 	Imports   string
 	Functions string
 }
+
+// HasEmptyInput checks to see if any required inputs were not filled in.
+func (inputs *ModuleInputs) HasEmptyInput() bool {
+	requiredInputs := []string{
+		inputs.ModuleName, inputs.Language, inputs.Namespace, inputs.ResourceType, inputs.ResourceSubtype, inputs.ModelName,
+	}
+	for _, input := range requiredInputs {
+		if input == "" {
+			return true
+		}
+	}
+	return false
+}
+
+// CheckResource checks if the given resource is valid.
+func (inputs *ModuleInputs) CheckResource() error {
+	resources := []string{
+		"arm component",
+		"audio_input component",
+		"base component",
+		"board component",
+		"camera component",
+		"encoder component",
+		"gantry component",
+		"generic component",
+		"gripper component",
+		"input component",
+		"motor component",
+		"movement_sensor component",
+		"pose_tracker component",
+		"power_sensor component",
+		"sensor component",
+		"servo component",
+		"generic service",
+		"mlmodel service",
+		"motion service",
+		"navigation service",
+		"slam service",
+		"vision service",
```

Comment:
I think these are all hard coded in `module_generate.go` as well, which means if anything changes there are two places we need to update and more points of failure/inconsistency. Can we instead put these in variables somewhere that both point to?

Reply:
I saw that as well, but wasn't sure how to really do this since they were being used in the form. I'll revisit this again!